debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -85,64 +92,48 @@ private PostgresReplicationConnection(Configuration config,
-        final String postgresPluginName = plugin.getPostgresPluginName();
-        ServerInfo.ReplicationSlot slotInfo;
+    private ServerInfo.ReplicationSlot getSlotInfo() throws SQLException, InterruptedException {
         try (PostgresConnection connection = new PostgresConnection(originalConfig)) {
-            slotInfo = connection.readReplicationSlotInfo(slotName, postgresPluginName);
+            return connection.readReplicationSlotInfo(slotName, plugin.getPostgresPluginName());
         }
+    }
+
+    private void ensureSlotNotActive(ServerInfo.ReplicationSlot slotInfo) throws IllegalStateException {
+        if (slotInfo.active()) {
+            throw new IllegalStateException(
+                    "A logical replication slot named '" + slotName + "' for plugin '" + plugin.getPostgresPluginName() + "' and database '" + database() + "' is already active on the server." +
+                    "You cannot have multiple slots with the same name active for the same database."
+            );
+        }
+    }
+
+    protected void initReplicationSlot() throws SQLException, InterruptedException {
+        ServerInfo.ReplicationSlot slotInfo = getSlotInfo();
 
         boolean shouldCreateSlot = ServerInfo.ReplicationSlot.INVALID == slotInfo;
         try {
             // there's no info for this plugin and slot so create a new slot
             if (shouldCreateSlot) {
-                LOGGER.debug("Creating new replication slot '{}' for plugin '{}'", slotName, plugin);
-
-                // creating a temporary slot if it should be dropped an we're on 10 or newer;
-                // this is not supported through the API yet
-                // see https://github.com/pgjdbc/pgjdbc/issues/1305
-                if (useTemporarySlot()) {
-                    try (Statement stmt = pgConnection().createStatement()) {
-                        stmt.execute(String.format(
-                                "CREATE_REPLICATION_SLOT %s TEMPORARY LOGICAL %s",
-                                slotName,
-                                postgresPluginName
-                        ));
-                    }
-                }
-                else {
-                    pgConnection().getReplicationAPI()
-                        .createReplicationSlot()
-                        .logical()
-                        .withSlotName(slotName)
-                        .withOutputPlugin(postgresPluginName)
-                        .make();
-                }
-            }
-            else if (slotInfo.active()) {
-                LOGGER.error(
-                        "A logical replication slot named '{}' for plugin '{}' and database '{}' is already active on the server." +
-                        "You cannot have multiple slots with the same name active for the same database",
-                        slotName, postgresPluginName, database());
-                throw new IllegalStateException();
+                this.createReplicationSlot();
```

Comment:
Why is this one needed here? Is this _only_ for the case where it hasn't been created before by the `RecordsSnapshotProducer`?

Reply:
yes, and I think also when temporary slots are used